### DISCUSSION TITLE
How can exchange the texture of a SpriteBundle at runtime?

### DISCUSSION THREAD

**Discussion by fabiandiez** [bevyengine/bevy#8054] [Q&A] [answered]
I want to be able to change the texture used by a sprite bundle from within a system at runtime. I tried searching trough existing Q&A threads, but all I could find was https://github.com/bevyengine/bevy/discussions/2892 which uses `Handle<ColorMaterial>` instead of the `Handle<Image>` that is being used in `SpriteBundle`. 
Am I supposed to use a `MaterialMesh2dBundle` instead? Or is there a way to make this working using `SpriteBundle`?

Any tips would be appreciated, thanks in advance.

**Comment by NiklasEi** [comment, marked as answer]
The answer of #2892 also applies here. You can change the value of the  `Handle<Image>` component.
Slightly adjusted code example from #2892:
```rust
fn update_sprite(asset_server: Res<AssetServer>, query: Query<&mut Handle<Image>, With<MyMarkerComponent>>) {
    let mut handle = query.single_mut();
    *handle = asset_server.load("sprites/new_sprite.png");
}
```

**Reply by fabiandiez** [reply]
Thanks for the quick response! Looks like it works now, thanks :)
